hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -166,12 +166,12 @@ public abstract class HoodieBackedTableMetadataWriter<I> implements HoodieTableM
    * @param failedWritesCleaningPolicy Cleaning policy on failed writes
    * @param engineContext              Engine context
    * @param inflightInstantTimestamp   Timestamp of any instant in progress
+   * @param shouldInitialize           Determines whether metadata writer should be initialized by reading the metadata
    */
-  protected HoodieBackedTableMetadataWriter(StorageConfiguration<?> storageConf,
-                                            HoodieWriteConfig writeConfig,
+  protected HoodieBackedTableMetadataWriter(StorageConfiguration<?> storageConf, HoodieWriteConfig writeConfig,
                                             HoodieFailedWritesCleaningPolicy failedWritesCleaningPolicy,
-                                            HoodieEngineContext engineContext,
-                                            Option<String> inflightInstantTimestamp) {
+                                            HoodieEngineContext engineContext, Option<String> inflightInstantTimestamp,
+                                            boolean shouldInitialize) {
```

Comment:
We should not be needing this flag. If the index is present in available metadata partitions in table config, then it is alrady initialized. Table config is always updated after completing metadata write. If index creation fails due to some reason, after completing metadata write but before updating table config, then it is sufficient to check the table config.

Reply:
Addressed